tls/s2n_handshake_io.c

```diff
@@ -418,6 +422,22 @@ static int s2n_advance_message(struct s2n_connection *conn)
         this_mode = 'C';
     }
 
+    /* If we were responding to a client hello with a retry request, reset the handshake */
+    if (conn->handshake.message_number == SERVER_HELLO && conn->handshake.server_sent_hrr == 1) {
```

Comment:
instead of relying on message_number, s2n_conn_get_current_message_type() should be used.

Reply:
Updated.